apps/dashboard/app/controllers/batch_connect/session_contexts_controller.rb

```diff
@@ -8,8 +8,12 @@ def new
     set_session_context
 
     if @app.valid?
-      # Read in context from cache file
-      @session_context.from_json(cache_file.read) if cache_file.file?
+      # Read in context from cache file if cache is disabled and context.json exist
+      begin
+       @session_context.update_with_cache(JSON.parse(cache_file.read))  if cache_file.file?
+      rescue => e
+        flash.now[:alert] = "Attempting to update form inputs with cached values resulted in a crash: #{e.message} "
+      end
```

Comment:
Seems like we should catch this at a lower layer, log the error, then continue and show the user default values without the cache. 

I mean, not being able to read the cache file shouldn't stop the app from working, it should just stop that particular feature.  I think we should employ graceful degradation.

Reply:
I'll address this outside of this PR.